compiler/ast/src/ast_gen.rs

```diff
@@ -10,14 +10,16 @@ type Ident = String;
 #[derive(Debug, PartialEq)]
 pub struct Located<T, U = ()> {
     pub location: Location,
+    pub end_location: Option<Location>,
     pub custom: U,
     pub node: T,
 }
 
 impl<T> Located<T> {
-    pub fn new(location: Location, node: T) -> Self {
+    pub fn new(location: Location, end_location: Location, node: T) -> Self {
```

Comment:
I kept `Location` as a required field in `::new` for now. (We can omit it in special cases by going through the Struct directly, as in `ast.rs`.)

Reply:
I could see arguments either way here. On the one hand, it's odd that the field you pass in differs from what's stored on the struct. On the other hand, this encourages good behavior _and_ reduces the number of code changes required.

Reply:
I agree this approach and the idea encouraging good behavior.